### PR TITLE
[consensus][e2e] Add e2e smoke test for changing onchain config and voting history fix

### PR DESCRIPTION
**Pull request by igor-aptos**
running locally with:
LOCAL_SWARM_NODE_RELEASE=1 cargo test --release --package smoke-test --lib -- aptos_cli::validator::test_onchain_config_change --exact --nocapture

we get:
With new config: 23 rounds from vote to elected: Some(20) to Some(43)
With old config: 61 rounds from vote to elected: Some(12) to Some(73)

This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4986)

### CODE REVIEW

**Review by davidiw — DISMISSED** [review]
Man, I misread PR... updating...

* we need a better framework for generating upgrades @runtian-zhou 
* it might want to use a proper governance process

had originally said stale... was not correct there

dismissing requested changes, but still might be good to clean up a bit. this is an update for next branch cut, right?

File: testsuite/smoke-test/src/aptos_cli/validator.rs [hunk 1]
```diff
@@ -76,6 +84,214 @@ async fn test_show_validator_set() {
...
+        "Epoch after : {}",
+        rest_client
+            .get_ledger_information()
+            .await
+            .unwrap()
+            .into_inner()
+            .epoch
+    );
+
+    // let txn = root_account.sign_with_transaction_builder(
```
Comment (davidiw) [inline testsuite/smoke-test/src/aptos_cli/validator.rs:253]: delete?

**Comment by igor-aptos** [comment]
Yeah this is a smoke test for the (already committed) leader election onchain config change planned for next release.

What I think we need to be able to do, to have smoke tests for governance proposals like these (which we really should have - otherwise they are not continuously tested):
- be able to set genesis onchain config. This basically tests a revert, because genesis is now new. I'll try to add this to smoke tests
- wrappers for generating proposals like these, and executing governance - i.e. we shouldn't need every smoke test to implement that
- be able to set smoke-tests to run nightly. this tests for example runs for multiple minutes, so not great to have it run on every PR :/

**Comment by runtian-zhou** [comment]
I'll include the consensus config in the proposal generator!

File: testsuite/smoke-test/src/aptos_cli/validator.rs [hunk 2]
```diff
@@ -76,6 +81,269 @@ async fn test_show_validator_set() {
...
+        .nth(2)
+        .map(|v| (v.peer_id(), v.rest_client()))
+        .unwrap();
+
+    rest_client_off
+        .set_failpoint("consensus::send::any".to_string(), "100%return".to_string())
+        .await
+        .unwrap();
+
+    // tokio::time::sleep(Duration::from_secs(20)).await;
```
Comment (bchocho) [inline testsuite/smoke-test/src/aptos_cli/validator.rs:105]: nit: stray comment :)

**Comment by igor-aptos** [comment]
Argh, I had these PRs stacked, merging this one doesn't merge in main @runtian-zhou . 

since this PR is marked as merged, I needed to recreate it - #5594

**Pull request merged**